OrbitLinuxTracing/TracerThread.cpp

```diff
@@ -828,6 +840,26 @@ void TracerThread::ProcessSampleEvent(const perf_event_header& header,
     thread_name.set_timestamp_ns(event->GetTimestamp());
     listener_->OnThreadName(std::move(thread_name));
 
+  } else if (!stream_id_to_tracepoint_.empty()) {
+    absl::flat_hash_map<uint64_t, orbit_grpc_protos::TracepointInfo>::iterator it =
+        stream_id_to_tracepoint_.begin();
+
+    while (it != stream_id_to_tracepoint_.end()) {
+      if (it->first == stream_id) {
+        auto event = ConsumeTracepointPerfEvent<TracepointEventPidTidTimeCpu>(ring_buffer, header);
+
+        orbit_grpc_protos::TracepointEvent tracepoint_event;
+        tracepoint_event.set_pid(event->GetPid());
+        tracepoint_event.set_tid(event->GetTid());
+        tracepoint_event.set_time(event->GetTimestamp());
+        tracepoint_event.set_cpu(event->GetCpu());
+
+        orbit_grpc_protos::TracepointInfo* tracepoint = tracepoint_event.mutable_tracepoint_info();
+        tracepoint->set_name(it->second.name());
+        tracepoint->set_category(it->second.category());
+      }
+      it++;
+    }
```

Comment:
This is missing the sending of the event.

Reply:
Yes, I reserved that for a future CL.